rmf_task/src/rmf_task/TaskPlanner.cpp

```diff
@@ -378,7 +378,12 @@ class TaskPlanner::Implementation
 
   ConstRequestPtr make_charging_request(rmf_traffic::Time start_time)
   {
-    return rmf_task::requests::ChargeBattery::make(start_time);
+    return rmf_task::requests::ChargeBattery::make(
+      start_time,
+      nullptr,
+      true,
+      std::nullopt,
```

Comment:
I'm thinking instead of passing `nullopt` for the requester, it might be good to define a `static constexpr std::string TASK_PLANNER_NAME = "task_planner";` either in the header file or inside the impl class. Then use this static variable everywhere we need to autogenerate tasks. This would allow auto injected tasks to also include the `requester` block in the task_state.json logs.